src/core/Object3D.js

```diff
@@ -303,35 +303,51 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 	lookAt: function () {
 
-		// This method does not support objects with rotated and/or translated parent(s)
+		// This method does not support objects having non-uniformly-scaled parent(s)
 
+		var q1 = new Quaternion();
 		var m1 = new Matrix4();
-		var vector = new Vector3();
+		var target = new Vector3();
+		var position = new Vector3();
 
 		return function lookAt( x, y, z ) {
 
 			if ( x.isVector3 ) {
 
-				vector.copy( x );
+				target.copy( x );
 
 			} else {
 
-				vector.set( x, y, z );
+				target.set( x, y, z );
 
 			}
 
+			var parent = this.parent;
+
+			this.updateWorldMatrix( true, false );
+
+			position.setFromMatrixPosition( this.matrixWorld );
+
 			if ( this.isCamera ) {
 
-				m1.lookAt( this.position, vector, this.up );
+				m1.lookAt( position, target, this.up );
 
 			} else {
 
-				m1.lookAt( vector, this.position, this.up );
+				m1.lookAt( target, position, this.up );
 
 			}
 
 			this.quaternion.setFromRotationMatrix( m1 );
 
+			if ( parent ) {
+
+				m1.extractRotation( parent.matrixWorld );
+				q1.setFromRotationMatrix( m1 );
+				this.quaternion.premultiply( q1.inverse() );
```

Comment:
Would not it be more efficient to compute the world quaternion of the parent by simply multiplying the ancestors quaternions? that may lift the non-uniformly-scaled contraint (to positively scaled?)

Reply:
Interesting...

Reply:
@WestLangley are you planning on taking a look at the suggestion?

Reply:
@mrdoob I have put it on my todo list. :-)

Reply:
Okay 😊

Reply:
@mbredif Extracting the quaternion from the world matrix is the correct approach. An ancestor's quaternion is not necessarily set, or even used (i.e., when `matrixAutoUpdate` is `false`).